src/internal/mod.rs

```diff
@@ -477,6 +485,52 @@ impl<'de> Deserialize<'de> for PrivateKey {
     }
 }
 
+impl PrivateKey {
+    /// Augment this private key with another, producing a new PrivateKey
+    fn augment(&self, augmenting_key: &AugmentationFactor) -> Result<PrivateKey, IronOxideErr> {
+        use recrypt::Revealed;
+        let zero: RecryptPrivateKey = RecryptPrivateKey::new([0u8; 32]);
+        if Revealed((augmenting_key.clone().0).0) == Revealed(zero.clone()) {
```

Comment:
Why does `Revealed` take ownership of its input, instead of borrowing? Seems like the clones shouldn't be necessary.

Reply:
I agree. See https://github.com/IronCoreLabs/recrypt-rs/issues/91